### PR TITLE
Np 1500 do not create doi request message without text

### PR DESCRIPTION
**Pull request by axthosarouris**
Fixed a bug where a DoiReqeustMessage would be created with null text.

### CODE REVIEW

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 1]
```diff
@@ -120,24 +120,19 @@ public DynamoDBDoiRequestsService(AmazonDynamoDB client,
     @Override
     public Optional<DoiRequestSummary> fetchDoiRequest(UUID publicationId) throws NotFoundException {
         Publication publication = fetchPublication(publicationId);
-        return Optional.ofNullable(DoiRequestSummary.fromPublication(publication));
+        return Optional.of(DoiRequestSummary.fromPublication(publication));
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:123]: Good catch.

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 2]
```diff
@@ -120,24 +120,19 @@ public DynamoDBDoiRequestsService(AmazonDynamoDB client,
...
     }
 
     @Override
     public void createDoiRequest(CreateDoiRequest createDoiRequest, String username)
```
Comment (norrs) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:127]: I believe we might want to refactor into `createDoiRequest` in `Publication` so we are telling the `Publication` object to create the doi request and it does the job and validation. 

Something that conflicts with my brain that it is `DynamoDBDoiRequestsService` doing the validation here.

Comment (axthosarouris) [reply doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:127]: I see the point of moving the business logic closer to the datamodel classes. But we should agree on a plan for this. We cannot just start moving functionality inside the datamodel classes. The reason we need a plan is that we have certain technical challenges. For example, if we move the createDoiRequest inside the Publication, we should also inject a clock in the Publication with one or another way in order to be able to set the create/update instants in a controlled way.  

I am not sure however, if this lies in the scope of this bugfix.

File: doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java [hunk 3]
```diff
@@ -150,6 +145,13 @@ public void updateDoiRequest(UUID publicationID, DoiRequestStatus requestedStatu
         putItem(publication);
     }
 
+    private Publication fetchPublicationForUser(CreateDoiRequest createDoiRequest, String username)
+        throws NotFoundException, ForbiddenException {
+        var publication = fetchPublication(UUID.fromString(createDoiRequest.getPublicationId()));
```
Comment (brinxmat) [inline doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java:150]: Worth noting that CreateDoiRequest::getPublicationId always gets called wrapped with UUID.fromString in usage contexts. Probably means this field has the type UUID. The handling of IllegalArgumentExceptions should probably be moved to commons parseBody as this is generic functionality.

**Review by brinxmat — APPROVED** [review]

**Review by janbj — APPROVED** [review]

**Pull request merged**